packages/react-aria-components/src/SearchField.tsx

```diff
@@ -55,6 +55,7 @@ function SearchField(props: SearchFieldProps, ref: ForwardedRef<HTMLDivElement>)
   let {validationBehavior: formValidationBehavior} = useSlottedContext(FormContext) || {};
   let validationBehavior = props.validationBehavior ?? formValidationBehavior ?? 'native';
   let inputRef = useRef<HTMLInputElement>(null);
+  let [inputContextProps, mergedInputRef] = useContextProps({}, inputRef, InputContext);
```

Comment:
Note this opens up the ability for the user to influence the TextField's input from outside the textfield now

Reply:
a bit weird, since this is the only place where we merge contexts instead of having the deepest one win, but otherwise we have to make `Autocomplete` provide the contexts for SearchField, TextField, TextArea, and all other possible fields which doesn't seem so sustainable...

Reply:
yeah, I was a bit torn here. I guess one down side of the current approach is that you'd need to remember to clear the context if there were input props from a higher level context that you didn't want...